### PR TITLE
Partial tallying

### PR DESCRIPTION
**Pull request by txus**
After receiving a quorum of trustee shares, if there are some missing we have the option to do partial tallying:

* Bulletin Board sends an empty `tally.compensate` message.
* Each trustee sends a `Compensation` in response to that message.
* Once the Bulletin Board gets all the `Compensation` messages it needs to proceed, it will emit the `end_tally` message with the decrypted tally as usual.

### CODE REVIEW

File: voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/common.py [hunk 1]
```diff
@@ -126,16 +149,17 @@ def process_message(self, message_type: str, message: Content) -> Content:
         return responses
 
     def is_fresh(self) -> bool:
-        return isinstance(self.step, self.starting_step)
+        return type(self.step) is self.__class__.starting_step
```
Comment (leio10) [inline voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/common.py:152]: 💬 Maybe we could make this method more explicitly. I think this should be only in the Trustee wrapper and we could set a variable to `False` after sending the election keys. What do you think?

File: voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/serializable.py [hunk 2]
```diff
@@ -59,27 +60,27 @@ def maybe_base64_to_int(i: Union[str, int]) -> int:
 
 
 def set_serializers():
-    old_set_serializers()
+    # old_set_serializers()
```
Comment (leio10) [inline voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/serializable.py:63]: ⚠️ should we keep this? if not, we should delete this commented code and the assignment of this function

File: voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/serializable.py [hunk 3]
```diff
@@ -59,27 +60,27 @@ def maybe_base64_to_int(i: Union[str, int]) -> int:
...
     )
     electionguard.serializable.set_serializer(
-        lambda i, **_: int_to_maybe_base64(i), int
+        lambda i, **_: int_to_maybe_base64(i), int  # type: ignore
     )
 
 
 def set_deserializers():
-    old_set_serializers()
+    # old_set_serializers()
```
Comment (leio10) [inline voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/serializable.py:76]: ⚠️ same as before. Also, the method name is wrong, so probably it was not doing anything 🙈

File: voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/bulletin_board.py [hunk 4]
```diff
@@ -1,46 +1,274 @@
...
-    serialize,
-    deserialize,
 )
-from .dummy_scheduler import DummyScheduler
+from .utils import InvalidBallot, deserialize, serialize
+
+TallyResults = Dict[CONTEST_ID, Dict[SELECTION_ID, int]]
+
+
+class TallyDecryptor:
```
Comment (leio10) [inline voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/bulletin_board.py:38]: 💬 This class is pretty big, maybe we could move it to a separate file?

**Review by leio10 — APPROVED** [review]
Awesome work!! 💪 👏

**Review by slickepinne — APPROVED** [review]
Really nice. I just left some minor notes, and I'm curious about the decryption when some guardians are missing. Would be amazing, if we can check it together when there's time. 😊

File: voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/serializable.py (hunk 2 above)
Comment (txus) [reply voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/serializable.py]: Kept it!

File: voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/bulletin_board.py (hunk 4 above)
Comment (txus) [reply voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/bulletin_board.py]: Extracted~

File: voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/serializable.py (hunk 3 above)
Comment (txus) [reply voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/serializable.py]: Kept it~

File: .code-workspace [hunk 5]
```diff
@@ -0,0 +1,32 @@
+{
+    "folders": [
+        {
+            "name": "Bulletin Board",
+            "path": "decidim-bulletin_board-app"
```
[the remark this answers is not in the document]
Comment (txus) [reply .code-workspace]: Cheers!

File: voting_schemes/electionguard/python-wrapper/Makefile [hunk 6]
```diff
@@ -18,13 +18,16 @@ install-apt:
 install-deps:
 	python3 -m pip install --upgrade pip
 	pip install pipenv
-	cd ../electionguard-python && make environment build
-	pipenv install --dev
-	python3 -m pip install --user --upgrade twine
+	PIPENV_VENV_IN_PROJECT=True pipenv install --dev
+	python -m pip install --user --upgrade twine
```
[the remark this answers is not in the document]
Comment (txus) [reply voting_schemes/electionguard/python-wrapper/Makefile]: Good one. cheers!

File: .code-workspace [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        },
+        {
+            "name": "Bulletin Board Gem",
+            "path": "decidim-bulletin_board-ruby"
```
[the remark this answers is not in the document]
Comment (txus) [reply .code-workspace]: Cheers!

File: voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/common.py (hunk 1 above)
Comment (txus) [reply voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/common.py:152]: I'm not sure about making it concrete to the Trustee. I think this is a general question you can ask about any wrapper --is it fresh or has it been restored from some backup?

**Pull request merged**